int64 nodeID
float64 joint_var

